_posts/2020-12-24-how-to-start-with-foreman

```diff
@@ -0,0 +1,399 @@
+Host represents some machine. It can be a device on network or simply something,
+that you want to install operating system. It's typically a thing you want to manage
+or automate some tasks around it.
+In virtualization terminology, it can be host but also a guest. It can be a baremetal
+machine, it can be VM running on some hypervisor or a VM somewhere in the cloud.
+While Foreman does not primarily target these, it can also be a router, IoT sensor,
+laptop, container.
+
+Regardless of what kind of machine the host is, Foreman calls that simply the Host. Hosts have name.
+They also must belong to some organization and location. This is the bare minimum
+that Foreman requires in order to define a Host. Some use-cases may require more
+attributes. E.g. to provision a host in the cloud, cloud credentials are required
+or in order to provision a host using PXE boot, host MAC address is needed. In order
+to run job though SSH on it, it's FQDN or IP must be set.
+
+Hosts can be either *managed* or *unmanaged*. While it's a simple boolean flag, it
+dramatically changes how Foreman treats them. Managed hosts are seen
+as something, that Foreman tries to manage in external services. Imagine you
+create a new Host in Foreman, you set it's name (FQDN), IP address, MAC address.
+Based on the managed flag, Foreman can reach out to associated DHCP server to create
+a reservation for the given MAC and IP pair. It would try to create A, AAAA and PTR records
+for the hostname and IP address pair. What exactly happens is driven by what is configured
+in Foreman. We won't go to more details in this post,
+but reading our [manual](https://theforeman.org/manuals/latest/#4.GeneralForeman)
+could give you some idea.
+
+For unmanaged hosts, Foreman does not reach out to external services
+and modify their configuration. The only exception to his rule is the
+communication to hypervisor/cloud in case of a virtual machine. Foreman
+would delete the linked VM when the Host is deleted from Foreman.
+
+Unmanaged Hosts can be used in many use cases. The first and most simple use case is the inventory.
+You can keep track of your infrastructure information. You can have multiple [users and control](https://theforeman.org/manuals/latest/index.html#4.1.2RolesandPermissions),
+who has access to specific subset of data. This data can be searched through
+using advanced, yet easy to use, search syntax. You can generate various reports
+and statistics. Therefore you can e.g. easily tell, how many RHEL 7 and 8 are
+running in your datacenter, search for all hosts in *Subnet* 192.168.0.1/24,
+create CSV list of all hosts that are owned by a specific group of admins.
+
+Once you have hosts defined, you can also take advantage of some nice plugins,
+such as [remote execution](https://www.theforeman.org/plugins/foreman_remote_execution/1.7/index.html)
+to run scripts and commands via SSH. Or run [ansible](https://www.theforeman.org/plugins/foreman_ansible/2.x/index.html)
```

Comment:
```suggestion
to run scripts and commands via SSH. Or run [Ansible](https://www.theforeman.org/plugins/foreman_ansible/2.x/index.html)
```

Reply:
also, a link to the 2.x docs, while we ship 6.x in 2.3? :)

Reply:
sadly we only have manual for 3.x, could infrateam help with copying the manual automatically for every plugin on release? the history shows us, maintainers can't keep up with manual work

Reply:
Could you elaborate how you would like to see that? A plain copy will be out of date since it's a major version. Maybe better for a separate discussion.